app/views/papers/_sort_by.html.erb

```diff
@@ -0,0 +1,4 @@
+<div class="btn-group btn-group-xs nav-button-group sort" role="group" aria-label="...">
+  <%= link_to 'Votes', url_for(sort: 'votes'), class: 'btn btn-default ' + ('active' if params[:sort] != 'pub_date').to_s, role: "button" %>
+  <%= link_to 'Publish Date', url_for(sort: 'pub_date'), class: 'btn btn-default '  + ('active' if params[:sort] == 'pub_date').to_s, role: 'button'  %>
```

Comment:
I did the to_s because it was throwing a "no implicit conversion of nil to
string" error, but I'll use the interpolation instead.  I'm using !=
pub_date to set the default as votes is active.  There might be a better
way of doing this like passing "sort: votes" initially or something.  I
could also do an unless, that will probably make it easier to read.

On Thu, Sep 8, 2016 at 7:29 PM, Brian Kung <notifications@github.com> wrote:

> In app/views/papers/_sort_by.html.erb
> <https://github.com/briankung/projectcredo/pull/70#discussion_r78112901>:
>
> > @@ -0,0 +1,4 @@
> > +<div class="btn-group btn-group-xs nav-button-group sort" role="group" aria-label="...">
> > +  <%= link_to 'Votes', url_for(sort: 'votes'), class: 'btn btn-default ' + ('active' if params[:sort] != 'pub_date').to_s, role: "button" %>
> > +  <%= link_to 'Publish Date', url_for(sort: 'pub_date'), class: 'btn btn-default '  + ('active' if params[:sort] == 'pub_date').to_s, role: 'button'  %>
>
> You can just use string interpolation here:
>
> "btn btn-default #{'active' if params[:sort] != 'votes}"
>
> Also the ('active' if params[:sort] != 'pub_date').to_s stuff is kind of
> silly for multiple reasons. #to_s is totally unnecessary, doubly so if
> you use string interpolation, and then checking if it's "not equal to
> 'pub_date'" is much less readable and has more side effects than just
> saying what you want - "make it active if the sort parameter is "votes."
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/briankung/projectcredo/pull/70/files/e5240cef8e29cf5c4de4f689dc3026f25a5f6af5#r78112901>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACKsZEO4YPjDRLnbmwQoWoYv1sa7ooOFks5qoKhzgaJpZM4J2i6I>
> .
>